src/jupyterhub_cost_monitoring/query_cost_aws.py

```diff
@@ -37,7 +37,9 @@ def _get_component_name(service_name):
         return "other"
 
 
-def query_aws_cost_explorer(metrics, granularity, from_date, to_date, filter, group_by):
+def query_aws_cost_explorer(
+    metrics, granularity, from_date, to_date, filter, group_by, aws_ce_client
```

Comment:
I guess we are adding the `aws_ce_client` just to enable injecting the mock client for tests? If that's the case, may be we should consider explicitly patching the client instead?

Reply:
Indeed, thank you for the suggestion below to patch the client directly – I'll check it out!